tests/direct_scraper/test_direct_scraper.py

```diff
@@ -5,29 +5,22 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import os
+import pytest
 
 from mozdownload import DirectScraper
 import mozdownload.errors as errors
-import mozhttpd_base_test as mhttpd
 from mozdownload.utils import urljoin
 
-
-class TestDirectScraper(mhttpd.MozHttpdBaseTest):
+@pytest.mark.parametrize('attr', ['binary', 'binary_regex', 'path', 'path_regex'])
+def test_url_download(httpd, tmpdir, attr):
     """test mozdownload direct url scraper"""
-
-    def test_url_download(self):
-        filename = 'download_test.txt'
-        test_url = urljoin(self.wdir, filename)
-        scraper = DirectScraper(url=test_url,
-                                destination=self.temp_dir,
-                                logger=self.logger)
-        self.assertEqual(scraper.url, test_url)
-        self.assertEqual(scraper.filename,
-                         os.path.join(self.temp_dir, filename))
-
-        for attr in ['binary', 'binary_regex', 'path', 'path_regex']:
-            self.assertRaises(errors.NotImplementedError, getattr, scraper, attr)
-
-        scraper.download()
-        self.assertTrue(os.path.isfile(os.path.join(self.temp_dir,
-                                                    scraper.filename)))
+    tmpdir = str(tmpdir)
+    filename = 'download_test.txt'
+    test_url = urljoin(httpd.get_url(), filename)
+    scraper = DirectScraper(url=test_url, destination=tmpdir)
+    assert scraper.url == test_url
+    assert scraper.filename == os.path.join(tmpdir, filename)
+    with pytest.raises(errors.NotImplementedError):
+        getattr(scraper, attr)
```

Comment:
If we move out this exception to a separate test then dont you think we need to extract scraper object which will be by following same procedure in two different functions for `assert os.path.isfile(os.path.join(str(tmpdir), scraper.filename))` and for `    with pytest.raises(errors.NotImplementedError):
        getattr(scraper, attr)` ?
we could run pytest.raises in a loop for it as a variable instead of parametrizing it.

Reply:
I agree separating exception would be better as decorator would run multiple times for the parametrized data in which part of code will be running again. I will make the asked changes on this.